source/manuals/programming-languages/nodejs/index.html.md.erb

```diff
@@ -253,6 +253,75 @@ const callback = filename => (err, data) => {
 };
 ```
 
+#### Use of `async`/`await`
+
+If your version of Node is above 7.10.0 you should use async functions, and the await keyword, to organise your asynchronous code. This will make it easier to read and write.
+
+```js
+// without async/await
+const getUserFromCreds = function (username, password) {
+  return getUserIdFromAuth(username, password)
+    .then(userId => {
+      return getUserFromApi(userId);
+    })
+    .then(user => {
+      return {
+        'id': user.id,
+        'name': user.name,
+        'age': user.ageInYears      
+      };
+    })
+    .catch(e => {
+      console.log(e);
+    });
+}
+
+// with async/await
+const getUserFromCreds = async function (username, password) {
+  try {
+    const userId = await getUserIdFromAuth(username, password);
+    const user = await getUserFromApi(userId);
+
+    return {
+      'id': user.id,
+      'name': user.name,
+      'age': user.ageInYears
+    };
+  } catch(e) {
+    console.log(e);
+  }
+}
+```
+
+The `await` keyword will pause execution until its promise fulfills. This means a sequence of awaited promises can only execute in the order they're declared.
+
+If you have multiple promises, you should try to [instantiate them separately from when you await them](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Async_await#The_downsides_of_asyncawait), otherwise each promise will be forced to wait for the previous one. Example:
+
+```js
+const getUserFromCreds = async function (username, password) {
+  try {
+    // use await to make this promise blocking
+    await userId = getUserIdFromAuth(username, password);
+
+    // start other promises running before awaiting
+    const avatarPromise = getAvatarForUser(userId);
+    const userPromise = getUserFromApi(userId);
+
+    const avatarImgUrl = await avatarPromise;
```

Comment:
Another option if you're waiting on multiple promises and want to make that clear is wrapping them in `Promise.all`. I think what you've got here is a clear example so not necessarily suggesting replacing it, just a useful pattern I've found in the past.

```js
const avatarRequest = getAvatarForUser(userId)
const userRequest = getUserFromApi(userId)

const [ avatarImgUrl, user ] = await Promise.all([ avatarRequest, userRequest ])
// return { ... }
```

Reply:
It makes a lot of sense to use what's available from the promises API first. In your example, do the promises returned by `getAvatarForUser` and `getUserFromApi` need to be stored in variables are could they be passed straight into `Promise.all`?

Reply:
@sfount I've had a go at rewriting that section in https://github.com/alphagov/gds-way/pull/445/commits/af00770b77430343bf1b80b4c922785de18d7f4e, be interested in your opinion.

Reply:
@tombye  looks great 👍 no need to store anything so passing the calls directly into the `await Promise.all` is totally valid. 

The link through to the static methods on Promises seems super helpful too.